src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Subscales/Subscales.utils.ts

```diff
@@ -0,0 +1,46 @@
+import { ActivitySettingsSubscale, Item } from 'shared/state';
+import { AnswerDTO, ElementType } from 'shared/types';
+
+import { formatActivityItemAnswers } from '../Report.utils';
+import { SubscaleToRender } from './Subscales.types';
+
+export const getSubscalesToRender = (
+  data: ActivitySettingsSubscale,
+  activityItems: Record<string, { answer: AnswerDTO; activityItem: Item }>,
+  subscalesObject: Record<string, ActivitySettingsSubscale>,
+  endDatetime: string,
+  result: SubscaleToRender,
+) => {
+  if (!data?.items || !result) {
+    return result;
+  }
+
+  for (const item of data.items) {
+    if (item.type === ElementType.Subscale) {
+      const nestedSubscale = getSubscalesToRender(
+        subscalesObject[item.name],
+        activityItems,
+        subscalesObject,
+        endDatetime,
+        result[data.name],
+      );
+
+      if (!result?.[data.name]) {
+        result[data.name] = nestedSubscale;
+      } else {
+        result[data.name] = {
```

Comment:
I didn't catch it, it's redundant action, isn't it?

Reply:
fixed